### PR TITLE
[GeoMechanicsApplication] Replaced a raw loop by using `std::transform`

### PR DESCRIPTION
**Pull request by avdg81**
**📝 Description**
Replaced a raw loop by using `std::transform`.

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_geometries/line_interface_geometry.h [hunk 1]
```diff
@@ -261,19 +261,21 @@ class LineInterfaceGeometry : public Geometry<Node>
...
-        for (std::size_t i = 0; i < number_of_midline_nodes; ++i) {
-            const auto mid_point = (points[i] + points[i + number_of_midline_nodes]) / 2;
-            result.push_back(make_intrusive<Node>(points[i].Id(), mid_point));
-        }
+        auto begin1         = points.ptr_begin();
+        auto begin2         = begin1 + number_of_midline_nodes;
+        auto make_mid_point = [](auto pPoint1, auto pPoint2) {
+            return make_intrusive<Node>(pPoint1->Id(), Point{(*pPoint1 + *pPoint2) / 2});
+        };
+        std::transform(begin1, begin2, begin2, result.ptr_begin(), make_mid_point);
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/custom_geometries/line_interface_geometry.h]: I really like that you managed to get this into a `std::transform`! The only thing I'm still struggling with a little bit is the naming of the iterators. I can't really think of better ones right now, but you have to read line 278 carefully to understand.

**Review by rfaasse — COMMENTED** [review]
Very nice improvement! I think the names of the iterators still might be improved (although I don't have a better suggestion already), but if we can't think of better names, we can still merge it in my opinion

File: applications/GeoMechanicsApplication/custom_geometries/line_interface_geometry.h (hunk 1 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/custom_geometries/line_interface_geometry.h]: Thank you so much for your comment. We agreed to rename `begin1` to `begin_of_first_side`, and `begin2` to `begin_of_second_side`.

**Review by rfaasse — APPROVED** [review]
Nice, good to go!

**Pull request merged**